bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -204,12 +208,27 @@ impl Client {
             tokio::sync::RwLock<Option<SessionVerificationController>>,
         > = Default::default();
         let controller = session_verification_controller.clone();
+        sdk_client.add_event_handler(move |event: AnyToDeviceEvent| async move {
+            if let AnyToDeviceEvent::KeyVerificationRequest(event) = event {
+                if let Some(session_verification_controller) = &*controller.clone().read().await {
+                    session_verification_controller
+                        .process_incoming_verification_request(
+                            &event.sender,
+                            event.content.transaction_id,
+                        )
+                        .await;
+                }
+            }
+        });
 
-        sdk_client.add_event_handler(move |ev: AnyToDeviceEvent| async move {
-            if let Some(session_verification_controller) = &*controller.clone().read().await {
-                session_verification_controller.process_to_device_message(ev).await;
-            } else {
-                debug!("received to-device message, but verification controller isn't ready");
+        let controller = session_verification_controller.clone();
+        sdk_client.add_event_handler(move |event: OriginalSyncRoomMessageEvent| async move {
```

Comment:
Before you ask, sadly I don't think we can make the handler here smarter like we did for the to-device variant.

Reply:
Ha, thank you for not letting me fumble around for nothing 😅